include/fmt/format.h

```diff
@@ -742,9 +742,14 @@ void basic_memory_buffer<T, SIZE, Allocator>::grow(size_t size) {
 #ifdef FMT_FUZZ
   if (size > 5000) throw std::runtime_error("fuzz mode - won't grow that much");
 #endif
+  const size_t max_size = std::allocator_traits<Allocator>::max_size(alloc_);
+  if (size > max_size) {
+    throw std::length_error("fmt::basic_memory_buffer::grow");
+  }
   size_t old_capacity = this->capacity();
   size_t new_capacity = old_capacity + old_capacity / 2;
-  if (size > new_capacity) new_capacity = size;
+  if (new_capacity > max_size) new_capacity = max_size;
+  if (size > new_capacity) new_capacity = size; // size is never greater max_size.
```

Comment:
We only need to check `new_capacity` if it is not greater than `size`. The rest `allocate` will take care of. Something like:

```
if (size > new_capacity) new_capacity = size;
else if (new_capacity > max_size) new_capacity = (std::max)(size, max_size);
```